liquibase-core/src/main/java/liquibase/integration/commandline/Main.java

```diff
@@ -1986,6 +1957,116 @@ protected void doMigration() throws Exception {
+                refUrl = value;
+            } else if (OPTIONS.REFERENCE_USERNAME.equalsIgnoreCase(attributeName)) {
+                refUsername = value;
+            } else if (OPTIONS.REFERENCE_PASSWORD.equalsIgnoreCase(attributeName)) {
+                refPassword = value;
+            } else if (OPTIONS.REFERENCE_DEFAULT_CATALOG_NAME.equalsIgnoreCase(attributeName)) {
+                refCatalogName = value;
+            } else if (OPTIONS.REFERENCE_DEFAULT_SCHEMA_NAME.equalsIgnoreCase(attributeName)) {
+                refSchemaName = value;
+            }
+//            } else if (OPTIONS.DATA_OUTPUT_DIRECTORY.equalsIgnoreCase(attributeName)) {
+//                dataOutputDirectory = value;
+//            }
+        }
+
+        if (refUrl == null) {
+            throw new CommandLineParsingException(
+                    String.format(coreBundle.getString("option.required"), "--referenceUrl"));
+        }
+
+        command.addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_DEFAULT_SCHEMA_NAME_ARG, refSchemaName)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_URL_ARG, refUrl)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_DEFAULT_CATALOG_NAME_ARG, refCatalogName)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_DRIVER_ARG, refDriver)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_DRIVER_PROPERTIES_FILE_ARG, null)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_USERNAME_ARG, refUsername)
+                .addArgumentValue(ReferenceDbUrlConnectionCommandStep.REFERENCE_PASSWORD_ARG, refPassword);
+    }
+
+    private ResourceAccessor getFileOpenerResourceAccessor() throws FileNotFoundException {
+        ResourceAccessor fileOpener;
+        if (Main.runningFromNewCli) {
+            fileOpener = Scope.getCurrentScope().getResourceAccessor();
+        } else {
+            fileOpener = new CompositeResourceAccessor(
+                    new DirectoryResourceAccessor(Paths.get(".").toAbsolutePath().toFile()),
+                    new ClassLoaderResourceAccessor(classLoader)
+            );
+        }
+        return fileOpener;
+    }
+
+    private DiffOutputControl getDiffOutputControl(CompareControl.SchemaComparison[] finalSchemaComparisons, ObjectChangeFilter objectChangeFilter) {
+        DiffOutputControl diffOutputControl = new DiffOutputControl(
+                includeCatalog, includeSchema, includeTablespace, finalSchemaComparisons);
+        for (CompareControl.SchemaComparison schema : finalSchemaComparisons) {
+            diffOutputControl.addIncludedSchema(schema.getReferenceSchema());
+            diffOutputControl.addIncludedSchema(schema.getComparisonSchema());
+        }
+
+        if (objectChangeFilter != null) {
+            diffOutputControl.setObjectChangeFilter(objectChangeFilter);
+        }
+
+        return diffOutputControl;
+    }
```

Comment:
Some javadoc on these methods would be a good idea.

Reply:
Good catch, I'll review it and add the javadocs